File: src/dataset/python/isbn-verifier/claude/ClaudeSonnet4_isbn-verifier_v3.py
def is_valid(isbn):
    clnisbn = isbn.replace('-', '')
    
    if len(clnisbn) != 10:
        return False
    
    if not (clnisbn[:-1].isdigit() and (clnisbn[-1].isdigit() or clnisbn[-1] == 'X')):
        return False
    
    sumisbn = sum(int(digit) * (10 - i) for i, digit in enumerate(clnisbn[:-1]))
    sumisbn += 10 if clnisbn[-1] == 'X' else int(clnisbn[-1])
    
    return sumisbn % 11 == 0 and sumisbn > 0

File: src/dataset/javascript/isbn-verifier_exercism-javascript-ThomasZumsteg/claude/ClaudeSonnet4_isbn-verifier_v3.js
export default class ISBN {
  constructor(digits) {
    this.digits = digits;
  }

  isValid() {
    const len = this.digits.length;
    if (len !== 10) return false;
    
    let sum = 0;
    let digit = 10;
    
    for (let i = 0; i < len; i++) {
      const char = this.digits[i];
      
      if (char >= '0' && char <= '9') {
        sum += digit * (char - '0');
        digit--;
      } else if (digit === 1 && char === 'X') {
        sum += 10;
        digit--;
      } else {
        return false;
      }
    }
    
    return digit === 0 && sum % 11 === 0;
  }
}